libs/@guardian/libs/src/storage/storage.ts

```diff
@@ -98,6 +98,32 @@ class StorageFactory {
 	setRaw(key: string, value: string): void {
 		return this.#storage?.setItem(key, value);
 	}
+
+	/**
+	 * Get key by index.
+	 * @returns the name of the key at that index (`string`), or `null` (if
+	 *      index is out of range of if storage is unavailable) - if the index
+	 *      is out of range, or if storage is not available.
+	 *
+	 * Wrapper for the `key` method on the native `Storage` object, with
+	 * additional check for availability of storage. Note that the _order_ of
+	 * keys in storage is dependent on user-agent implementation, and so you
+	 * should not assume that keys will be stored in any particular order (e.g.
+	 * order of insertion).
+	 * @see https://developer.mozilla.org/en-US/docs/Web/API/Storage/key
+	 */
+	key(index: number): string | null {
+		return this.#storage?.key(index) ?? null;
+	}
+
+	/**
+	 * Get the number of items in storage.
+	 * @returns the number of items in storage, or `undefined` if storage is
```

Comment:
I've had the `key` method return `null` if storage is unavailable, because the native method can return `null`, and it seemed best not to introduce a third possible return type. But it generally feels more semantic to return `undefined` if the storage object isn't available so I've used `undefined` for that in the `length` method.

Reply:
I would actually return `null` or `undefined` everywhere. While I favour `undefined` in general, I think using `null` here would be more consistent.

Reply:
okay thanks, happy to make this change!